src/commands/publish/link-publish-mode.ts

```diff
@@ -54,20 +62,68 @@ export const linkPublishMode = async (
 		return;
 	}
 
-	/**
-	 * If it's a symlink, make sure it's in the node_modules directory of the base package.
-	 * e.g. This could happen with pnpm
-	 *
-	 * If it's not, it might be a development directory and we don't want to overwrite it.
-	 */
-	const linkPathReal = await fs.realpath(linkPath);
-	if (!linkPathReal.startsWith(expectedPrefix)) {
-		return;
-	}
-
-	await hardlinkPackage(
+	const throttledHardlinkPackage = throttle(hardlinkPackage, 500);
+	await throttledHardlinkPackage(
 		linkPath,
 		absoluteLinkPackagePath,
 		packageJson,
 	);
+
+	if (watchMode) {
+		const globOptions = {
+			globstar: true,
+			extended: true,
+		};
+
+		/**
+		 * npm-packlist ignore list:
+		 * https://github.com/npm/npm-packlist/blob/v8.0.2/lib/index.js#L15-L38
+		 */
+		const ignoreFiles = [
+			// Files
+			'**/{npm-debug.log,*.orig,package-lock.json,yarn.lock,pnpm-lock.yaml}',
+
+			// Folders
+			'**/node_modules/**',
+
+			// Hidden files
+			'**/.{_*,*.swp,DS_Store,gitignore,npmrc,npmignore,lock-wscript,.wafpickle-*}',
+
+			// Hidden folders
+			'**/.{_*,git,svn,hg,CVS}/**',
+		].map(glob => globToRegexp(glob, globOptions));
+
+		const watcher = fs.watch(
+			absoluteLinkPackagePath,
+			{ recursive: true },
+		);
+
+		for await (const { eventType, filename } of watcher) {
+			if (!filename) {
+				continue;
+			}
+
+			const shouldIgnore = ignoreFiles.some(ignoreFile => ignoreFile.test(filename));
```

Comment:
Probably doesn't matter too much, but one thing you could consider if you're sticking with `globToRegexp` is joining the regexes together then calling test just once rather than in a loop:

```diff
diff --git a/src/commands/publish/link-publish-mode.ts b/src/commands/publish/link-publish-mode.ts
index e0b535c..69caad3 100644
--- a/src/commands/publish/link-publish-mode.ts
+++ b/src/commands/publish/link-publish-mode.ts
@@ -79,7 +79,7 @@ export const linkPublishMode = async (
 		 * npm-packlist ignore list:
 		 * https://github.com/npm/npm-packlist/blob/v8.0.2/lib/index.js#L15-L38
 		 */
-		const ignoreFiles = [
+		const ignoreFiles = new RegExp([
 			// Files
 			'**/{npm-debug.log,*.orig,package-lock.json,yarn.lock,pnpm-lock.yaml}',
 
@@ -91,7 +91,8 @@ export const linkPublishMode = async (
 
 			// Hidden folders
 			'**/.{_*,git,svn,hg,CVS}/**',
-		].map(glob => globToRegexp(glob, globOptions));
+		].map(glob => globToRegexp(glob, globOptions).source)
+		 .join('|'));
 
 		const watcher = fs.watch(
 			absoluteLinkPackagePath,
@@ -103,7 +104,7 @@ export const linkPublishMode = async (
 				continue;
 			}
 
-			const shouldIgnore = ignoreFiles.some(ignoreFile => ignoreFile.test(filename));
+			const shouldIgnore = ignoreFiles.test(filename);
 			if (shouldIgnore) {
 				continue;
 			}

```